cmd/kubeadm/app/apis/output/types.go

```diff
@@ -82,3 +84,43 @@ type UpgradePlan struct {
 
 	ConfigVersions []ComponentConfigVersionState
 }
+
+// Certificate represents information about check-expiration for one certificate
+type Certificate struct {
+	// Name of the certificate
+	Name string
+	// ExpirationDate defines certificate expiration date.
+	ExpirationDate metav1.Time
+	// ResidualTime represents the time missing to expiration
+	ResidualTime time.Duration
+	// ExternallyManaged defines if the certificate is externally managed
+	IsExternallyManaged bool
+	// CAName represents the name of the CA that signed the certificate
+	CAName string
+	// IsMissing represents if the certificate is missing
+	IsMissing bool
+}
+
+// CertificateAuthority represents information about check-expiration for one certificate authority
+type CertificateAuthority struct {
```

Comment:
let's use a single structure only called Certficate.
the field CAName can be optional if the cert is a CA without a parent CA.

comment:
```
Certificate represents information for a certificate or a certificate authority when using the check-expiration command.
type Certificate struct {
...
```

Reply:
NOTE: please apply the same suggested changes for the v1alpha4/types.go